federation/pkg/typeadapters/secret.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package typeadapters
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	federationclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_clientset"
+	"k8s.io/kubernetes/federation/pkg/federation-controller/util"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+)
+
+type SecretAdapter struct {
+	client federationclientset.Interface
+}
+
+func NewSecretAdapter(client federationclientset.Interface) *SecretAdapter {
+	return &SecretAdapter{client: client}
+}
+
+func (a *SecretAdapter) SetClient(client federationclientset.Interface) {
+	a.client = client
+}
+
+func (a *SecretAdapter) Kind() string {
+	return "secret"
+}
+
+func (a *SecretAdapter) Equivalent(obj1, obj2 pkgruntime.Object) bool {
+	secret1 := obj1.(*apiv1.Secret)
+	secret2 := obj2.(*apiv1.Secret)
+	return util.SecretEquivalent(*secret1, *secret2)
```

Comment:
Yes, I agree.  For now the comparison method is shared with the unit and e2e testing but as those are refactored that need will likely go away.